### ISSUE TITLE
fix: hydrated bloc for integration tests

### ISSUE THREAD

**Issue by dbilgin** [felangel/bloc#3078]
Hi there, I am having some problems with my integration tests with hydrated_bloc, I was hoping someone could help me out here. I run my app under main like this:

final storage = await HydratedStorage.build(
    storageDirectory: await getApplicationDocumentsDirectory(),
);
HydratedBlocOverrides.runZoned(
      () => runApp(MyApp()),
      storage: storage,
)

I saw no problems running the app regularly.
I have these integration tests that initialize the application like this:

import 'package:my_app/main.dart' as app;

Future<void> _startApp(WidgetTester tester) async {
  getIt.reset();
  await app.main();
  await tester.pumpAndSettle();
}

When I run this code I get the error:

StorageNotFound (Storage was accessed before it was initialized.
Please ensure that storage has been initialized.

For example:

final storage = await HydratedStorage.build();
HydratedBlocOverrides.runZoned(
  () => runApp(MyApp()),
  storage: storage,
);)

Although I obviously do this. When I run my tests on debug mode, they fail immediately like this, but when I run them  regularly, they fail on the second test. I'm quite confused about why this might be happening.

**Comment by felangel** [comment]
Hi @dbilgin 👋
Thanks for opening an issue!

Can you please provide a link to a minimal reproduction sample? It would be much easier to help if we are able to reproduce the issue locally, thanks! 🙏

**Comment by JanKubicek** [comment]
Hi @felangel,

I've been experiencing the same issue and I have created repo where you can reproduce it [https://github.com/JanKubicek/hydrated_bloc_integration_test_example](url). In the readme file you can see how to run the tests. If you change HydratedBloc to Bloc, integration tests work.

Thanks!

**Comment by dbilgin** [comment]
Thanks @JanKubicek for setting that up, I didn't have time initially.
I can confirm I am having the same problem as in your example project. Just had to add an await [here](https://github.com/JanKubicek/hydrated_bloc_integration_test_example/blob/master/integration_test/app_test.dart#L14).

**Comment by yonjans** [comment]
I am experiencing same issue. Package Version: `8.0.0`.

Any plan for this?

**Comment by felangel** [comment]
> I am experiencing same issue. Package Version: `8.0.0`.
> 
> Any plan for this?

Yup I’m working on a fix. Should be available sometime today/tomorrow.

**Comment by BellRampion** [comment]
Has the patch been released yet? I am also experiencing this issue. Code works fine pre-8.0.0; upgrading causes HydratedBloc to throw the storage error.

**Comment by felangel** [comment]
> Has the patch been released yet? I am also experiencing this issue. Code works fine pre-8.0.0; upgrading causes HydratedBloc to throw the storage error.

I’m still working on it and will post and update either today or tomorrow. Sorry for the inconvenience!

**Comment by felangel** [comment]
This is blocked on https://github.com/flutter/flutter/issues/96939 at the moment :(

**Comment by dbilgin** [comment]
Thanks for taking a look at it @felangel! Hopefully it gets fixed soon.

**Comment by felangel** [comment]
@jswietek this should be resolved now via v9.0.0-dev.1 of hydrated_bloc. You can check out the https://github.com/felangel/bloc/tree/sample/e2e branch for an example. Let me know if that helps 👍
